rust/lance-index/src/traits.rs

```diff
@@ -77,15 +88,5 @@ pub trait DatasetIndexExt {
     async fn optimize_indices(&mut self) -> Result<()>;
 
     /// Find index with a given index_name and return its serialized statistics.
-    async fn index_statistics(&self, index_name: &str) -> Result<Option<String>>;
-
-    /// Count the rows that are not indexed by the given index.
-    ///
-    /// TODO: move to [DatasetInternalExt]
-    async fn count_unindexed_rows(&self, index_name: &str) -> Result<Option<usize>>;
-
-    /// Count the rows that are indexed by the given index.
-    ///
-    /// TODO: move to [DatasetInternalExt]
-    async fn count_indexed_rows(&self, index_name: &str) -> Result<Option<usize>>;
+    async fn index_statistics(&self, index_name: &str) -> Result<serde_json::Value>;
```

Comment:
This puts `serde_json` on our public API.  It's just a minor pain (e.g. users have to make sure they are using the same version of `serde_json`).  Is there a particular reason you didn't want to use `String` here?  Maybe for composing statistics from child indices?

I don't think this is something we need to fix for this PR, just noting it.  Maybe we can just make a follow-up issue for it.